structuralcodes/codes/mc2010/_concrete_shear.py

```diff
@@ -0,0 +1,1598 @@
+            fck, bw, z, E_s, As, Med, Ved, Ned, delta_e, alfa, gamma_c
+        )
+    raise ValueError("invalid approx level")
+
+
+def v_rd_max_approx1(
+    fck: float,
+    bw: float,
+    theta: float,
+    z: float,
+    alfa: float = 0,
+    gamma_c: float = 1.5,
+) -> float:
+    """The maximum allowed shear resistance, when there is shear reinforcment
+
+    fib Model Code 2010, eq. (7.3-37)
+
+    Args:
+        fck (float): Characteristic strength in MPa
+        z: (float): The length to the areasenter of cross-section in mm
+        bw: (float): Thickness of web in cross section
+        delta_e (float): The exentricity of the load in mm
+        alfa (float): Inclination of the stirrups in degrees
+        gamma_c (float): Safety factor
+
+    Returns:
+        float: The maximum allowed shear resistance regardless of
+        approximation level"""
+    nfc = min((30 / fck) ** (1 / 3), 1)
+
+    return (
+        0.55
+        * nfc
+        * (fck / gamma_c)
+        * bw
+        * z
+        * (
+            ((1 / tan(theta * pi / 180)) + (1 / tan(alfa * pi / 180)))
+            / (1 + (1 / tan(theta * pi / 180)) ** 2)
+        )
+    )
+
+
+def v_rd_max_approx2(
+    fck: float,
+    bw: float,
+    theta: float,
+    z: float,
+    E_s: float,
+    As: float,
+    Med: float,
+    Ved: float,
+    Ned: float,
+    delta_e: float,
+    alfa: float = 0,
+    gamma_c: float = 1.5,
+) -> float:
+    """The maximum allowed shear resistance, when there is shear reinforcment
+
+    fib Model Code 2010, eq. (7.3-24), (7.3-40) and (7.3-41)
+
+    Args:
+        fck (float): Characteristic strength in MPa
+        z: (float): The length to the areasenter of cross-section in mm
+        bw: (float): Thickness of web in cross section
+        dg: (float): Maximum size of aggregate
+        E_s: (float): The E_s-modulus to the materialb in MPa
+        As: (float): The cross-section area of reinforcement in mm^2
+        Med: (float): The moment working on the material in Nmm
+        Ved: (float): The shear working on the material in N
+        Ned: (float): The normal force working on the material in N
+        delta_e (float): The exentricity of the load in mm
+        alfa (float): Inclination of the stirrups in degrees
+        gamma_c (float): Concrete safety factor
+
+    Returns:
+        float: The maximum allowed shear resistance regardless of
+        approximation level"""
+    nfc = min((30 / fck) ** (1 / 3), 1)
+
+    epsilon_1 = epsilon_x(E_s, As, Med, Ved, Ned, z, delta_e) + (
+        epsilon_x(E_s, As, Med, Ved, Ned, z, delta_e) + 0.002
+    ) * ((1 / tan(theta * pi / 180)) ** 2)
+    k_epsilon = min(1 / (1.2 + 55 * epsilon_1), 0.65)
+
+    return (
+        k_epsilon
+        * nfc
+        * (fck / gamma_c)
+        * bw
+        * z
+        * (
+            ((1 / tan(theta * pi / 180)) + (1 / tan(alfa * pi / 180)))
+            / (1 + (1 / tan(theta * pi / 180)) ** 2)
+        )
+    )
+
+
+def v_rd_max_approx3(
```

Comment:
Is there a need to re-do all these calculations? Possible to just make a call to `v_rd_max_approx2`?